backend/routes/create_active_order.py

```diff
@@ -1,24 +1,165 @@
+            #     error_code=400,
+            #     error_message="User not found",
+            # )
+            return "User not found"
+        if user_profile["balance"] < price * quantity:
+            # raise_http_exception(
+            #     status_code=HTTP_400_BAD_REQUEST,
+            #     error_code=400,
+            #     error_message="Insufficient balance",
+            # )
+            return "Insufficient balance"
+    else:  # Sell order
+        # portfolio = supabase_middleman.fetch_portfolio(user_id, stock_id)
+        portfolio = supabase_middleman.get_user_portfolio(user_id)[stock_id]
+        if not portfolio:
+            # raise_http_exception(
+            #     status_code=HTTP_400_BAD_REQUEST,
+            #     error_code=400,
+            #     error_message="Portfolio not found",
+            # )
+            return "Stock not found in users portfolio"
+        if portfolio["quantity"] < 1:
+            # raise_http_exception(
+            #     status_code=HTTP_400_BAD_REQUEST,
+            #     error_code=400,
+            #     error_message="User does not own stock",
+            # )
+            return "User does not own stock"
+        if portfolio["quantity"] < quantity:
+            # raise_http_exception(
+            #     status_code=HTTP_400_BAD_REQUEST,
+            #     error_code=400,
+            #     error_message="Insufficient quantity",
+            # )
+            return "Insufficient quantity"
+
+    time_posted = datetime.now().isoformat()
+    # create_active_order.create_active_order(
+    #     time_posted, buy_or_sell, price, expirey, quantity, stock_id, user_id
+    # )
+
+    generated_order_id = str(uuid4())
 
-    Returns: None
-    """
-    test_entry = test_data.test_entry_1()
-    supabase_middleman.insert_entry("active_buy_sell", test_entry)
+    if buy_or_sell:
+        supabase_middleman.update_user_balance(user_id, price * quantity * -1)
+        # supabase_middleman.escrow_funds(user_id, price, quantity)
+    else:
+        supabase_middleman.update_user_portfolio(user_id, stock_id, quantity * -1)
+        # supabase_middleman.escrow_stock(user_id, stock_id, quantity)
 
+    expirey = str(expirey)
+    entries = []
+    # Generate an entry for each quantity and insert those entries into the table
+    for i in range(quantity):
+        entry = {
+            "time_posted": time_posted,
+            "buy_or_sell": buy_or_sell,
+            "price": price,
+            "expirey": expirey,
+            "quantity": 1,
+            "stockId": stock_id,
+            "userId": user_id,
+            "orderId": generated_order_id,
+            "has_been_processed": False,
+        }
+        entries.append(entry)
+        # Insert the entry into the active_buy_sell table
+        # supabase_middleman.insert_entry("active_buy_sell", entry)
+    supabase_middleman.insert_entry("active_buy_sell", entries)
+    return "Active order created"
```

Comment:
Use dictionary unpacking instead:

```suggestion
    entries = [
        {
            "time_posted": time_posted,
            "buy_or_sell": buy_or_sell,
            "price": price,
            "expiry": str(expiry),
            "quantity": 1,
            "stockId": stock_id,
            "userId": user_id,
            "orderId": generated_order_id,
            "has_been_processed": False,
        }
        for _ in range(quantity)
    ]
    supabase_middleman.insert_entry("active_buy_sell", entries)
    return "Active order created"
```

Reply:
That is a pretty clever trick I didnt know existed. Merged in.